src/lib/core/src/identity/player_identity.rs

```diff
@@ -9,3 +17,30 @@ impl PlayerIdentity {
         Self { username, uuid }
     }
 }
+
+fn uuid_to_u128_encoded_uuid<'de, D>(deserializer: D) -> Result<u128, D::Error>
+where
+    D: serde::Deserializer<'de>,
+{
+    let uuid_str: String = Deserialize::deserialize(deserializer)?;
+    let id = uuid::Uuid::try_parse(&uuid_str)
+        .map_err(|e| serde::de::Error::custom(format!("failed to parse UUID: {e}")))?;
+    let bytes = id.as_bytes();
+    let mut u = 0u128;
+    for &b in bytes {
+        u = (u << 8) | b as u128;
```

Comment:
just wondering why you didn't use u128::from_le_bytes (i.e. (u128::from_le_bytes(id.as_bytes().try_into().unwrap())
(i don't know the concrete types and am prettty new to rust)
in encoded_u128_uuid_to_uuid_str too; why did you not use the u128 function for converting u128s?